### PR TITLE
New Windows implementation using c# and proxytoggle

### PR DESCRIPTION
**Pull request by myleshorton**
The windows implementation uses the code from here:

https://github.com/getlantern/proxytoggle

### CODE REVIEW

File: sysproxy_darwin.go [hunk 1]
```diff
@@ -14,7 +14,7 @@ func ensureElevatedOnDarwin(be *byteexec.Exec, prompt string, iconFullPath strin
 	var s syscall.Stat_t
 	// we just checked its existence, not bother checking specific error again
 	if err = syscall.Stat(be.Filename, &s); err != nil {
-		return fmt.Errorf("Error stating helper tool %s: %s", err)
+		return fmt.Errorf("Error stating helper tool %s: %s", be.Filename, err)
```
Comment (myleshorton) [inline sysproxy_darwin.go]: linter

File: sysproxy.go [hunk 2]
```diff
@@ -34,12 +33,7 @@ func EnsureHelperToolPresent(path string, prompt string, iconFullPath string) (e
 	assetName := "sysproxy"
 	// Load different binaries for 32bit and 64bit Windows respectively.
 	if runtime.GOOS == "windows" {
-		suffix := "_386.exe"
-		// https://blogs.msdn.microsoft.com/david.wang/2006/03/27/howto-detect-process-bitness/
-		if strings.EqualFold(os.Getenv("PROCESSOR_ARCHITECTURE"), "amd64") ||
-			strings.EqualFold(os.Getenv("PROCESSOR_ARCHITEW6432"), "amd64") {
-			suffix = "_amd64.exe"
-		}
+		suffix := ".exe"
```
Comment (myleshorton) [inline sysproxy.go]: The csharp implementation is a universal binary.

**Comment by myleshorton** [comment]
@oxtoacart this just switches to a c# version of toggling the proxy, which is just a heck of a lot simpler code-wise but also avoids using unmanaged memory, is a much smaller binary, and is a universal binary.

More than anything, though, I think this will be less error prone and more maintainable going forward, and that in general we should move towards c# of f# implementations on windows wherever possible (although I have no idea if you can target old .net runtimes with f#).

**Comment by myleshorton** [comment]
FYI @joesis. This effectively kills XP support AFAIK, although XP is only 3.56% of all users. Those users are also likely to be low-revenue.

**Comment by joesis** [comment]
When I running https://github.com/getlantern/proxytoggle/blob/master/proxytoggle.exe?raw=true on Windows 10 VM, the OS prompts me to install .NET framework 3.5. Windows 8 to 10 has .NET framework 4.5 to 4.6 respectively, but not 3.5.

As per https://msdn.microsoft.com/library/hh506443.aspx, there's probably no single .NET framework fits all OS version.

> In general, do not uninstall any versions of the .NET Framework from your computer. Different apps depend on different versions of the framework and multiple versions of the .NET Framework can be loaded on a single computer at the same time.

**Comment by joesis** [comment]
On my Win 7 VM, proxytoggle.exe does work correctly.

**Comment by joesis** [comment]
After installing .NET 3.5, proxytoggle.exe works on my Win 10 VM. It feels to me that the dependency is too heavy for such a simple utility.

**Comment by oxtoacart** [comment]
> It feels to me that the dependency is too heavy for such a simple utility.

Agreed. If we can't get this to work without some users being stuck having to install a specific dot net framework, I don't think it's worth it.

FWIW, the sysproxy-cmd.exe seems to be working fine now, and I suspect that just like pac-cmd.exe before it, we'll rarely if ever touch it again.

**Comment by myleshorton** [comment]
Ah, interesting point on the .net versions -- that's crazy they don't make the runtimes automatically backwards compatible. Hmn...I'll look at that. Thanks @joesis

**Comment by myleshorton** [comment]
OK I'm going to close this for now guys given all the other things we have to work on. In theory it would be cool to detect .net versions installed on user's machines in analytics since that's much more relevant for this than OS, but we can hold off for now. Thanks for giving it a look!

**Pull request closed without merging**